backend/src/routes/__tests__/card.spec.js

```diff
@@ -0,0 +1,529 @@
+        done();
+      })
+      .catch(done);
+  });
+  
+  it('PATCH /cards/:id as Unlogged should return 401', (done) => {
+    const updatedCard = {
+      name: 'New name',
+      description: 'New description',
+      image: 'https://via.placeholder.com/150'
+    };
+    request(app)
+      .patch(`/cards/${cardId}`)
+      .send(updatedCard)
+      .expect(401)
+      .then((response) => {
+        expect(response.body.reason).toBe(undefined);
+        done();
+      })
+      .catch(done);
+  });
+
+  it('PATCH /cards/:id should return 400 if name is empty', (done) => {
+    request(app)
+      .patch(`/cards/${cardId}`)
+      .set('Authorization', `Bearer ${adminToken}`)
+      .send({ ...card, name: '' })
+      .expect(400)
+      .expect('Content-Type', /json/)
+      .then((response) => {
+        expect(response.body.invalidFields).toContain('name');
+        done();
+      })
+      .catch(done);
+  });
+
+  it('PATCH /cards/:id should return 200 if name is not provided', (done) => {
+    const noTitleCard = {...card};
+    delete noTitleCard.name;
+    request(app)
+      .patch(`/cards/${cardId}`)
+      .set('Authorization', `Bearer ${adminToken}`)
+      .send(noTitleCard)
+      .expect(200)
+      .expect('Content-Type', /json/)
+      .then((response) => {
+        expect(response.body.invalidFields).not.toBeDefined();
+        done();
+      })
+      .catch(done);
+  });
+});
+
+
+describe('Updating a Card PUT', () => {
+  it('PUT /cards/:id as an Admin should update the card', (done) => {
+    const newCard = {
+      ...updatedCard,
+      id: cardId,
+    };
+    request(app)
+      .put(`/cards/${cardId}`)
+      .set('Authorization', `Bearer ${adminToken}`)
+      .send(newCard)
+      .expect(200)
+      .expect('Content-Type', /json/)
+      .then((response) => {
+        expect(response.body.name).toBe(updatedCard.name);
+        expect(response.body.description).toBe(updatedCard.description);
+        expect(response.body.image).toBe(updatedCard.image);
+        expect(response.body.type).toBe(updatedCard.type);
+        expect(response.body.rarity).toBe(updatedCard.rarity);
+        expect(response.body.cost).toBe(updatedCard.cost);
+        expect(response.body.attack).toBe(updatedCard.attack);
+        expect(response.body.health).toBe(updatedCard.health);
+        expect(typeof new Date(response.body.createdAt).toISOString()).toBe('string');
+        expect(response.body.createdAt).toBeDefined();
+        expect(typeof new Date(response.body.updatedAt).toISOString()).toBe('string');
+        expect(response.body.updatedAt).toBeDefined();
+        done();
+      })
+      .catch(done);
+  });
+
+  it('PUT /cards/:id as a Player should return 403', (done) => {
+    request(app)
+      .put(`/cards/${cardId}`)
+      .set('Authorization', `Bearer ${playerToken}`)
+      .send(updatedCard)
+      .expect(403)
+      .then((response) => {
+        expect(response.body.reason).toBe(undefined);
```

Comment:
Il faut plutot vérifier que la réponse entière n'est pas un object.

expect(typeof response.body).not.toBe('object');